src/chord.cpp

```diff
@@ -415,6 +415,50 @@ bool Chord::HasNoteWithDots()
     return false;
 }
 
+void Chord::AdjustOverlappingLayers(Doc *doc, const std::vector<LayerElement *> &otherElements, bool &isUnison)
+{
+    int margin = 0;
+    // get positions of other elements
+    std::set<int> otherElementLocations;
+    for (auto element : otherElements) {
+        if (element->Is(NOTE)) {
+            Note *note = vrv_cast<Note *>(element);
+            assert(note);
+            otherElementLocations.insert(note->GetDrawingLoc());
+        }
+    }
+    // get current chord positions
+    std::set<int> chordElementLocations;
+    for (int i = 0; i < GetChildCount(NOTE); ++i) {
+        Note *note = vrv_cast<Note *>(GetChild(i, NOTE));
+        assert(note);
+        chordElementLocations.insert(note->GetDrawingLoc());
+    }
+    const int expectedElementsInUnison
+        = CountElementsInUnison(chordElementLocations, otherElementLocations, GetDrawingStemDir());
+    const bool isLowerPosition = (STEMDIRECTION_down == GetDrawingStemDir()
+        && (*chordElementLocations.begin() >= *otherElementLocations.begin()));
+    int actualElementsInUnison = 0;
+    // process each note of the chord separately, storing locations in the set
+    for (int i = 0; i < GetChildCount(NOTE); ++i) {
+        Note *note = vrv_cast<Note *>(GetChild(i, NOTE));
+        assert(note);
+        auto [overlap, isInUnison]
+            = note->CalcNoteHorizontalOverlap(doc, otherElements, true, isLowerPosition, expectedElementsInUnison > 0);
+        if (((margin >= 0) && (overlap > margin)) || ((margin <= 0) && (overlap < margin))) {
+            margin = overlap;
+        }
+        if (isInUnison) ++actualElementsInUnison;
+    }
```

Comment:
No, not a problem. Actually this would be an improvement for the code. Initial implementation was relying on the indices of notes in the chord, but that is long since gone.
I'll update it to use GetList() instead.